### ISSUE TITLE
Notification and Lock screen controls are not displayed.

### ISSUE THREAD

**Issue by andrewoid** [googlecast/CastCompanionLibrary-android#192]
In the onResume() of [VideoCastControllerFragment.java](https://github.com/googlecast/CastCompanionLibrary-android/blob/master/src/com/google/android/libraries/cast/companionlibrary/cast/player/VideoCastControllerFragment.java) on line 525 the UI counter is incremented 

`mCastManager.incrementUiCounter();`

It is important that incrementUiCounter() always has a matching decrementUiCounter().

However, it is possible to leave the onResume() without incrementUiCounter() being called (see reproducible steps). Two possible code paths are
1. An Exception is thrown in onResume() before line 525.
2. The Activity is closed and return is called on line 519.

Recommendation: move the incrementUiCounter() out of the try-catch to after super.onResume() on line 506.

Reproducible Steps of code path 1 using [CastVideos-android](https://github.com/googlecast/CastVideos-android)

(these steps were reported to us by Google for our app)
1. Launch the app
2. Cast a video and **open the expanded controls**
3. Lock the phone
4. On Lollipop device press X on the lock screen controls to tear down the cast session
5. Unlock the phone.
6. **You are still inside VideoCastControllerActivity however media isn't being cast.**
7. Navigate Back.
8. Connect to the receiver and cast a video
9. Lock the phone

Expected Results: 
Lock screen controls should appear.

Actual results: 
Lock screen controls do not appear. Lock screen controls will not appear until the app is killed and reopened after following the above steps.

Additional Note:
Code path 1 is not reproducible on v1.14 because onResume() was different.

**Comment by naddaf** [comment]
That change is fine and makes sense (so I will make that change shortly) but what I am not fully understanding is that if the increment method is not called (due to exception, etc), then the effect would be reverse, i.e. notification would stay on more than it should (since, via increment, we potentially hide the notification) so how is that causing notification not to show up?

**Comment by andrewoid** [comment]
Notifications get displayed if the uiCounter == 0.
What should happen is that when we returned to the Activity (after Chromecast had been disconnected) the uiCounter should've been incremented to 1 (in onResume()) and then decremented to 0 in onPause() when the Activity closed. The parent Activity would increment uiCounter and everything should be fine.
But when we returned to the Activity the increment was never called in onResume(). When the decrement is called in onPause() the uiCounter is now -1. When we go back to the parent Activity it increments the uiCounter in it's onResume(), but we don't see the notification because the uiVisible flag is set to true. When we leave the Activity the uiCounter is decremented to -1 but since the Notification is only shown if uiCounter == 0, then no Notifications are ever shown.
